lib/ipmi_quantaoem.c

```diff
@@ -93,10 +96,10 @@ ipmi_get_qct_platform_id(struct ipmi_intf *intf)
 		lprintf(LOG_ERR, "Get Platform ID command failed: %#x %s", rsp->ccode, val2str(rsp->ccode, completion_code_vals));
 		return 0;
 	}
-	oem_id = rsp->data[0];
-	lprintf(LOG_DEBUG,"Platform ID: %x", oem_id);
+	platform_id = rsp->data[0];
+	lprintf(LOG_DEBUG,"Platform ID: %x", platform_id);
 
-	return oem_id;
+	return platform_id;
```

Comment:
You're still returning an unsigned byte up-converted to unsigned int16. What's the purpose?
Besides, in fact you're apparently returning your freshly defined enum type, `oem_qct_platform_id_t` (or `Code_Base` in your committed version). Please change the function type to that. Also, the `0` that you return above to indicate an error seems to be a valid platform_id. Namely, it's GRANTLEY. So you can't use 0. You need to define some `OEM_QCT_PLATFORM_UNKNOWN` value in your enum and return it.

Please pay more attention when processing the review comments.